### PR TITLE
update-js-sdk

### PR DESCRIPTION
**Pull request by nenadnovovic-ab**
## Summary by CodeRabbit

- **New Features**
	- Introduced new interfaces for evaluation results, enhancing data encapsulation.
	- Added methods to retrieve detailed information about local and CI test runs and their evaluations.

- **Tests**
	- Implemented new test cases to validate the functionality of the added methods in the `AutoblocksAPIClient`, ensuring comprehensive coverage for local and CI runs and results.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
   timeout?: TimeDelta;
 }
 
+interface ResultWithEvaluations {
```
Comment (nenadnovovic-ab) [inline src/client.ts:141]: @adamnolte I'm a bit uncertain about these types

File: src/client.ts [hunk 2]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
...
+  output: {
+    id: string;
+    name: string;
+    value: string;
+    contentType: HumanReviewFieldContentType;
+  }[];
+  evaluations: {
+    evaluatorId: string;
+    score: number;
+    passed: boolean | null;
```
Comment (adamnolte) [inline src/client.ts:163]: Let's make sure what is returned from the public api reflects this
```suggestion
    passed?: boolean;
```

File: src/client.ts [hunk 3]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
...
+    name: string;
+    value: string;
+    contentType: HumanReviewFieldContentType;
+  }[];
+  output: {
+    id: string;
+    name: string;
+    value: string;
+    contentType: HumanReviewFieldContentType;
+  }[];
```
Comment (adamnolte) [inline src/client.ts:159]: we should make these as unknown and let folks pass in types for them using generics in the get results method

File: src/client.ts [hunk 4]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
...
+  evaluations: {
+    evaluatorId: string;
+    score: number;
+    passed: boolean | null;
+    threshold?: number;
```
Comment (adamnolte) [inline src/client.ts:164]:
```suggestion
    threshold?: {
  lt?: number;
  lte?: number;
  gt?: number;
  gte?: number;
};
```

File: src/client.ts [hunk 5]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
   timeout?: TimeDelta;
 }
 
+interface ResultWithEvaluations {
+  id: string;
+  runId: string;
+  hash: string;
+  datasetItemId: string;
```
Comment (adamnolte) [inline src/client.ts:145]:
```suggestion
  datasetItemId?: string;
```

File: src/client.ts [hunk 6]
```diff
@@ -138,6 +138,34 @@ interface ClientArgs {
...
+  durationMs: number;
```
Comment (adamnolte) [inline src/client.ts:146]:
```suggestion
  durationMs?: number;
```

File: src/client.ts [hunk 7]
```diff
@@ -265,6 +293,58 @@ export class AutoblocksAPIClient {
     );
   }
 
+  public async getLocalRuns(testExternalId: string): Promise<{
```
Comment (adamnolte) [inline src/client.ts:296]: let's prefix all the methods here with Test so

getLocalTestRuns

getCITestRuns

GetLocalTestResults

and so on...

**Review by adamnolte — APPROVED** [review]

File: src/client.ts [hunk 8]
```diff
@@ -265,6 +288,64 @@ export class AutoblocksAPIClient {
...
+  public async getCITestResultWithEvaluations<
+    ResultBodyType = unknown,
+    ResultOutputType = unknown,
+  >(
+    testCaseResultId: string,
+  ): Promise<ResultWithEvaluations<ResultBodyType, ResultOutputType>> {
+    return this.get(
+      `/testing/ci/results/${encodeURIComponent(testCaseResultId)}`,
+    );
+  }
```
Comment (adamnolte) [inline src/client.ts:347]: Let's actually rename these and drop the `WithEvaluations`

File: src/client.ts [hunk 9]
```diff
@@ -138,6 +138,29 @@ interface ClientArgs {
   timeout?: TimeDelta;
 }
 
+interface ResultWithEvaluations<T = unknown, U = unknown> {
```
Comment (adamnolte) [inline src/client.ts:141]:
```suggestion
interface TestResult<T = unknown, U = unknown> {
```

**Review by pavle-autoblocks — APPROVED** [review]

**Pull request merged**